third_party/txt/src/txt/paragraph.cc

```diff
@@ -318,20 +319,68 @@ void Paragraph::Layout(double width, bool force) {
 
     size_t layout_start = run.start;
     // Layout until the end of the run or too many lines.
-    while (layout_start < run.end && lines_ < paragraph_style_.max_lines) {
+    while (layout_start < run.end && lines_ < max_lines) {
       const size_t next_break = (break_index > breaks_count - 1)
                                     ? std::numeric_limits<size_t>::max()
                                     : breaks[break_index];
       const size_t layout_end = std::min(run.end, next_break);
 
       bool bidiFlags = paragraph_style_.rtl;
+      std::shared_ptr<minikin::FontCollection> minikin_font_collection =
```

Comment:
Can you invoke this accessor only if you decide ellipsizing is necessary (within the `if` block).

Reply:
The font collection is also used by the call to layout.doLayout.  Updated the doLayout call accordingly